src/defiat/api.ts

```diff
@@ -456,6 +456,26 @@ export const getPoolApr = async (
   pid: number
 ) => {
   try {
+    const latestDistributionBlock = new BigNumber(
+      await Vault.methods.lastDistributionBlock().call()
+    )
+    const rewardsDistributed = await Vault.getPastEvents("RewardsDistributed", {
+      fromBlock: latestDistributionBlock.minus(50000),
+      toBlock: latestDistributionBlock,
+    });
```

Comment:
This is such an epic change. Never thought of using the events to do this.

If we start getting NaNs on the FE, we can reduce the block total. This function will only return a max of 10K results. Doubt it will happen though, so would like to keep this for now.